pkg/controller/certificatesigningrequest/ipsec_csr_approver.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package certificatesigningrequest
+
+import (
+	"context"
+	"crypto/x509"
+	"fmt"
+	"reflect"
+	"strings"
+
+	certificatesv1 "k8s.io/api/certificates/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	sautil "k8s.io/apiserver/pkg/authentication/serviceaccount"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+
+	antreaapis "antrea.io/antrea/pkg/apis"
+	"antrea.io/antrea/pkg/util/env"
+)
+
+const (
+	ipsecCSRApproverName = "AntreaIPsecCSRApprover"
+)
+
+var (
+	antreaAgentServiceAccountName = strings.Join([]string{
+		"system", "serviceaccount", env.GetAntreaNamespace(), "antrea-agent",
+	}, ":")
+)
+
+type ipsecCSRApprover struct {
+	client clientset.Interface
+}
+
+var ipsecTunnelUsages = sets.NewString(
+	string(certificatesv1.UsageIPsecTunnel),
+)
+
+var _ approver = (*ipsecCSRApprover)(nil)
+
+func (ic *ipsecCSRApprover) recognize(csr *certificatesv1.CertificateSigningRequest) bool {
+	if csr.Spec.SignerName != antreaapis.AntreaIPsecCSRSignerName {
+		return false
+	}
+	for _, usage := range csr.Spec.Usages {
+		if !ipsecTunnelUsages.Has(string(usage)) {
```

Comment:
I think this should be part of `verify` since the signer name is already this approver.
We should probably deny the request explicitly if a CSR for this signer declares unintended usage.